support/vm/testing.go

```diff
@@ -0,0 +1,276 @@
+package vm_test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/abi/big"
+	"github.com/filecoin-project/specs-actors/actors/builtin"
+	"github.com/filecoin-project/specs-actors/actors/builtin/account"
+	"github.com/filecoin-project/specs-actors/actors/builtin/cron"
+	"github.com/filecoin-project/specs-actors/actors/builtin/exported"
+	initactor "github.com/filecoin-project/specs-actors/actors/builtin/init"
+	"github.com/filecoin-project/specs-actors/actors/builtin/market"
+	"github.com/filecoin-project/specs-actors/actors/builtin/power"
+	"github.com/filecoin-project/specs-actors/actors/builtin/reward"
+	"github.com/filecoin-project/specs-actors/actors/builtin/system"
+	"github.com/filecoin-project/specs-actors/actors/builtin/verifreg"
+	"github.com/filecoin-project/specs-actors/actors/runtime/exitcode"
+	"github.com/filecoin-project/specs-actors/actors/util/adt"
+	"github.com/filecoin-project/specs-actors/support/ipld"
+	actor_testing "github.com/filecoin-project/specs-actors/support/testing"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/specs-actors/actors/runtime"
+	"github.com/ipfs/go-cid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	cbg "github.com/whyrusleeping/cbor-gen"
+)
+
+var FIL = big.NewInt(1e18)
+
+//
+// Genesis like setup
+//
+
+// Creates a new VM and initializes all singleton actors plus a root verifier account.
+func NewVMWithSingletons(ctx context.Context, t *testing.T) *VM {
+	store := ipld.NewADTStore(ctx)
+
+	lookup := map[cid.Cid]exported.BuiltinActor{}
+	for _, ba := range exported.BuiltinActors() {
+		lookup[ba.Code()] = ba
+	}
+
+	vm := NewVM(ctx, lookup, store)
+
+	emptyMapCID, err := adt.MakeEmptyMap(vm.store).Root()
+	require.NoError(t, err)
+	emptyArrayCID, err := adt.MakeEmptyArray(vm.store).Root()
+	require.NoError(t, err)
+	emptyMultimapCID, err := adt.MakeEmptyMultimap(vm.store).Root()
+	require.NoError(t, err)
+
+	initializeActor(ctx, t, vm, &system.State{}, builtin.SystemActorCodeID, builtin.SystemActorAddr, big.Zero())
+
+	// Verified registry account needs to be instantiated and initialized in init actor prior to construction
+	// of verified registry actor.
+	rootVerifier := verifregRootAddresses(t)
+	addrTreeRoot := initAddressTree(t, vm, rootVerifier)
+	initState := initactor.ConstructState(addrTreeRoot, "scenarios")
+	initState.NextID = abi.ActorID(builtin.FirstNonSingletonActorId)
+	initializeActor(ctx, t, vm, initState, builtin.InitActorCodeID, builtin.InitActorAddr, big.Zero())
+
+	rewardState := reward.ConstructState(big.Mul(big.NewInt(100), FIL))
```

Comment:
The argument is a power value, so multiplying by FIL is very weird. Add a new constant for `Petabyte` or similar.